client/app/services/query.js

```diff
@@ -293,6 +294,21 @@ function QueryResource($resource, $http, $q, $location, currentUser, QueryResult
     return this.getParameters().get();
   };
 
+  Query.prototype.getLocalParametersDefs = function getLocalParametersDefs() {
+    if (!this.$localParameters) {
+      this.$localParameters = this.getParametersDefs().filter(p => !p.global);
```

Comment:
It's safe to cache because it's used only in dashboards, where the query won't change - right?

Maybe we should move these two functions to the dashboard code, to avoid someone using it "by accident" in another context?